examples/tensor/tensor_construction.cpp

```diff
@@ -0,0 +1,151 @@
+//
+//  Copyright (c) 2018-2019, Cem Bassoy, cem.bassoy@gmail.com
+//
+//  Distributed under the Boost Software License, Version 1.0. (See
+//  accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+//  The authors gratefully acknowledge the support of
+//  Fraunhofer IOSB, Ettlingen, Germany
+//
+
+
+#include <boost/numeric/ublas/tensor.hpp>
+#include <iostream>
+#include <type_traits>
+
+int main() {
+    
+    using namespace boost::numeric::ublas;
+
+    auto t1 = tensor<float>{1,2,3,4};
+    auto t2 = tensor<float,dynamic_extents<4>>{dynamic_extents<4>{1,2,3,4}};
+    auto t3 = tensor<float,dynamic_extents<>>{dynamic_extents<>{1,2,3,4}};
+    auto t5 = tensor<float,dynamic_extents<>>{dynamic_extents<>{1,2,3,4},0.1}; 
+    auto t4 = tensor<float,static_extents<1,2,3,4>>();
+
+    auto t6 = tensor<float,static_extents<1,2,3,4>,first_order>();
+    auto t7 = tensor<float,dynamic_extents<>,first_order>{1,2,3,4};
+
+    auto t8 = tensor<float,dynamic_extents<>,first_order,std::vector<float>>{1,2,3,4};
+    auto t9 = tensor<float,static_extents<1,2,3,4>,first_order,std::array<float,24>>();
+
+    auto t1_e = t1.extents();
+    auto t2_e = t2.extents();
+    auto t3_e = t3.extents();
+    auto t4_e = t4.extents();
+    auto t5_e = t5.extents();
+    auto t6_e = t6.extents();
+    auto t7_e = t7.extents();
+    auto t8_e = t8.extents();
+    auto t9_e = t9.extents();
+
+    auto t1_s = t1.strides();
+    auto t2_s = t2.strides();
+    auto t3_s = t3.strides();
+    auto t4_s = t4.strides();
+    auto t5_s = t5.strides();
+    auto t6_s = t6.strides();
+    auto t7_s = t7.strides();
+    auto t8_s = t8.strides();
+    auto t9_s = t9.strides();
+
+    puts("\nt1: ");
```

Comment:
out of curiosity: why do you use `puts()` here rather than `std::cout` ?

Reply:
I think [Jason Turner](https://www.youtube.com/watch?v=VZTVmKOXLVU) could explain it better than me. If you would still like to know I could try. `puts` adds new-line character at the end by itself so I was trying to save a few characters and more about it you can learn from `Jason Turner` talked about in detail.

Reply:
Well, if the saving of a few characters is the reason, I'd argue this is a bad idea. You are using two distinct APIs and I/O frameworks to generate output. And while the implementers of the C++ standard library try really hard to keep the C++ API compatible with the C API, it's not always possible. For example, consider someone wanting to hook into the iostreams to redirect (or filter) the output stream. He would only see the data sent through `std::cout`, but not `puts()`.
In short, there are quite a number of ramifications to consider, and just to save a few key strokes  isn't worth it.

Reply:
Agree with Stefan here.

Reply:
Resolve conversation as file is outdated.